src/amo/reducers/addonsByAuthors.js

```diff
@@ -78,32 +83,86 @@ type LoadAddonsByAuthorsAction = {|
 |};
 
 export const loadAddonsByAuthors = (
-  { addons, forAddonSlug }: LoadAddonsByAuthorsParams
+  { addons, addonType, authors, forAddonSlug }: LoadAddonsByAuthorsParams
 ): LoadAddonsByAuthorsAction => {
   invariant(addons, 'A set of add-ons is required.');
+  invariant(authors, 'A list of authors is required.');
 
   return {
     type: LOAD_ADDONS_BY_AUTHORS,
-    payload: { addons, forAddonSlug },
+    payload: { addons, addonType, authors, forAddonSlug },
   };
 };
 
-export const getAddonsForSlug = (state: State, slug: string) => {
+export const joinAuthorNames = (
+  authorNames: Array<string>, addonType?: string
+) => {
+  return authorNames.sort().join('-') + (addonType ? `-${addonType}` : '');
```

Comment:
We do in general, but here I think it would look worse. We end up with:

```js
return `${authorNames.sort().join('-')}${(addonType ? `-${addonType}` : ''}`;
```

is that even valid? Because we want a nested template string. Maybe ES6 allows that but it looks less comprehensible to me.

Reply:
Yeah, I figured the reason you used `+` was because the template string would get messy. I'm fine with it, just wanted to check.